pkg/cli/cmd/credential/register/aws/accesskey/accesskey.go

```diff
@@ -0,0 +1,165 @@
+)
+
+func NewCommand(factory framework.Factory) (*cobra.Command, framework.Runner) {
+	runner := NewRunner(factory)
+
+	cmd := &cobra.Command{
+		Use:   "access-key",
+		Short: "Register (Add or update) AWS cloud provider credential for a Radius installation.",
+		Long: `Register (Add or update) AWS cloud provider credential for a Radius installation..
+
+This command is intended for scripting or advanced use-cases. See 'rad init' for a user-friendly way
+to configure these settings.
+
+Radius will use the provided IAM credential for all interactions with AWS. 
+` + common.LongDescriptionBlurb,
+		Example: `
+# Register (Add or update) cloud provider credential for AWS with IAM authentication
+rad credential register aws access-key --access-key-id <access-key-id> --secret-access-key <secret-access-key>
+`,
+		Args: cobra.ExactArgs(0),
+		RunE: framework.RunCommand(runner),
+	}
+
+	commonflags.AddOutputFlag(cmd)
+	commonflags.AddWorkspaceFlag(cmd)
+
+	cmd.Flags().String("access-key-id", "", "The AWS IAM access key id.")
+	_ = cmd.MarkFlagRequired("access-key-id")
+
+	cmd.Flags().String("secret-access-key", "", "The AWS IAM secret access key.")
+	_ = cmd.MarkFlagRequired("secret-access-key")
+
+	return cmd, runner
+}
+
+// Runner is the runner implementation for the `rad credential register aws` command.
+type Runner struct {
+	ConfigHolder      *framework.ConfigHolder
+	ConnectionFactory connections.Factory
+	Output            output.Interface
+	Format            string
+	Workspace         *workspaces.Workspace
+
+	AccessKeyID     string
+	SecretAccessKey string
+	KubeContext     string
+}
+
+// NewRunner creates a new instance of the `rad credential register aws` runner.
+func NewRunner(factory framework.Factory) *Runner {
+	return &Runner{
+		ConfigHolder:      factory.GetConfigHolder(),
+		ConnectionFactory: factory.GetConnectionFactory(),
+		Output:            factory.GetOutput(),
+	}
+}
+
+// Validate runs validation for the `rad credential register aws` command.
+//
+
+// Validate() checks if the required workspace, output format, access key ID and secret access key are present, and if not, returns an error.
+func (r *Runner) Validate(cmd *cobra.Command, args []string) error {
+	workspace, err := cli.RequireWorkspace(cmd, r.ConfigHolder.Config, r.ConfigHolder.DirectoryConfig)
+	if err != nil {
+		return err
+	}
+	r.Workspace = workspace
+
+	format, err := cli.RequireOutput(cmd)
+	if err != nil {
+		return err
+	}
+	r.Format = format
+
+	accessKeyID, err := cmd.Flags().GetString("access-key-id")
+	if err != nil {
+		return err
+	}
+	secretAccessKey, err := cmd.Flags().GetString("secret-access-key")
+	if err != nil {
+		return err
+	}
+	r.AccessKeyID = accessKeyID
+	r.SecretAccessKey = secretAccessKey
+
+	if r.AccessKeyID == "" {
+		return clierrors.Message("Access Key id %q cannot be empty.", r.AccessKeyID)
+	}
+	if r.SecretAccessKey == "" {
+		return clierrors.Message("Secret Access Key %q cannot be empty.", r.SecretAccessKey)
```

Comment:
Why do we need to print the empty `AccessKey` and `AccessKeyID` here? I mean why do we need `%q` here?

Reply:
this is the exisitng code for access-key type. we moved it to different file